File: java/demo/IceGrid/icebox/HelloI.java
// **********************************************************************
//
// Copyright (c) 2003-2014 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

import Demo.*;

public class HelloI extends _HelloDisp
{
    HelloI(String serviceName)
    {
        _serviceName = serviceName;
    }

    public void
    sayHello(Ice.Current current)
    {
        java.util.Map<String,String> env =  System.getenv();
        String lang = env.containsKey("LANG") ? env.get("LANG") : "en";
        String greeting = "Hello, ";
        if(lang.equals("fr"))
        {
            greeting = "Bonjour, ";
        }
        else if(lang.equals("de"))
        {
            greeting = "Hallo, ";
        }
        else if(lang.equals("es"))
        {
            greeting = "Hola, ";
        }
        else if(lang.equals("it"))
        {
            greeting = "Ciao, ";
        }        
        System.out.println(greeting + _serviceName);
    }

    private String _serviceName;
}

File: cs/demo/IceGrid/icebox/HelloI.cs
// **********************************************************************
//
// Copyright (c) 2003-2013 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

using Demo;

public class HelloI : HelloDisp_
{
    public HelloI(string serviceName)
    {
        _serviceName = serviceName;
    }

    public override void sayHello(Ice.Current current)
    {
#if COMPACT || SILVERLIGHT
        string lang = "en";
#else
        string lang = System.Environment.GetEnvironmentVariable("LANG") != null ? 
                                                            System.Environment.GetEnvironmentVariable("LANG") : "en";
#endif
        string greeting = "Hello, ";
        if(lang.Equals("fr"))
        {
            greeting = "Bonjour, ";
        }
        else if(lang.Equals("de"))
        {
            greeting = "Hallo, ";
        }
        else if(lang.Equals("es"))
        {
            greeting = "Hola, ";
        }
        else if(lang.Equals("it"))
        {
            greeting = "Ciao, ";
        }
        System.Console.Out.WriteLine(greeting + _serviceName);
    }

    private string _serviceName;
}
